src/network/clientpackethandler.cpp

```diff
@@ -922,44 +920,36 @@ void Client::handleCommand_AddParticleSpawner(NetworkPacket* pkt)
 {
 	u16 amount;
 	float spawntime;
-	v3f minpos;
-	v3f maxpos;
-	v3f minvel;
-	v3f maxvel;
-	v3f minacc;
-	v3f maxacc;
-	float minexptime;
-	float maxexptime;
-	float minsize;
-	float maxsize;
+	v3f minpos, maxpos;
+	v3f minvel, maxvel;
+	v3f minacc, maxacc;
+	float minexptime, maxexptime;
+	float minsize, maxsize;
 	bool collisiondetection;
 	u32 server_id;
+	bool vertical;
+	bool collision_removal;
+	u16 attached_id;
+	u8 glow;
 
 	*pkt >> amount >> spawntime >> minpos >> maxpos >> minvel >> maxvel
 		>> minacc >> maxacc >> minexptime >> maxexptime >> minsize
 		>> maxsize >> collisiondetection;
 
 	std::string texture = pkt->readLongString();
 
-	*pkt >> server_id;
+	*pkt >> server_id >> vertical >> collision_removal >> attached_id;
+
 
-	bool vertical = false;
-	bool collision_removal = false;
 	TileAnimationParams animation;
 	animation.type = TAT_NONE;
-	u8 glow = 0;
-	u16 attached_id = 0;
-	try {
-		*pkt >> vertical;
-		*pkt >> collision_removal;
-		*pkt >> attached_id;
-
+	{
 		// This is horrible but required (why are there two ways to deserialize pkts?)
 		std::string datastring(pkt->getRemainingString(), pkt->getRemainingBytes());
 		std::istringstream is(datastring, std::ios_base::binary);
 		animation.deSerialize(is, m_proto_ver);
-		glow = readU8(is);
-	} catch (...) {}
+	}
+	*pkt >> glow;
```

Comment:
I don't think this will work, glow needs to be read from the data that is left after the `TileAnimationParams` were read